.github/workflows/precommit.yml

```diff
@@ -246,6 +246,62 @@ jobs:
             .tox/.coverage
             .tox/coverage.xml
 
+  test-ast-encoding:
+    name: Test AST Encoding py-${{ matrix.os }}-${{ matrix.python-version }}
+    needs: build
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, windows-latest, ubuntu-latest]
+        # TODO SNOW-1763185 use py39 by default
+        # Protobuf encoding differs per python version, use 3.8 by default.
```

Comment:
We switched away from the base64 encoding, and are currently using textproto.

@sfc-gh-lspiegelberg Is this comment still relevant? Can we use Python 3.9 instead?

Reply:
we can, if we do not rely on python3.8 toolchain server-side (bazel). If we want to use bazel for phase1 in any capacity, I'd say let's be conservative and keep 3.8. 

Tbh, I see this as a minor detail. There's already a ticket so we can do a one-liner updating PR. I don't expect breakage from python 3.8 -> python 3.9 client-side.

Reply:
Let's try it on the private branch first then.